src/lnp/application/message.rs

```diff
@@ -591,6 +634,237 @@ pub struct ChannelReestablish {
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct ChannelUpdate {
+    /// Signature
+    pub signature: Signature,
+
+    /// Chainhash
+    pub chain_hash: AssetId,
+
+    /// Short Channel Id
+    pub short_channel_id: ShortChannelId,
+
+    /// Time stamp
+    pub timestamp: u32,
+
+    /// message flags
+    pub message_flags: u8,
+
+    /// channel flags
+    pub channle_flags: u8,
+
+    /// cltv expiry delta
+    pub cltv_expiry_delta: u16,
+
+    /// minimum HTLC in msat
+    pub htlc_minimum_msal: u64,
+
+    /// base fee in msat
+    pub fee_base_msat: u32,
+
+    /// fee proportional millionth
+    pub fee_proportional_millionths: u32,
+
+    /// if option_channel_htlc_max is set
+    pub htlc_maximum_msat: u64,
+}
+
+/// Extended Gossip messages
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct QueryShortChannelIds {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// short ids to query
+    pub short_ids: Vec<ShortChannelId>,
+
+    /// short id tlv stream
+    #[tlv(type = 1)]
+    pub short_id_tlvs: BTreeMap<u8, Vec<u8>>, /* is this the correct way to
+                                               * write tlv streams? */
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct ReplyShortChannelIdsEnd {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// full information
+    pub full_information: u8,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct QueryChannelRange {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// first block number
+    pub first_block_num: u32,
+
+    /// number of blocks
+    pub number_of_blocks: u32,
+
+    /// channel range queries
+    #[tlv(type = 1)]
+    pub query_channel_range_tlvs: BTreeMap<u8, Vec<u8>>,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct ReplyChannelRange {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// first block number
+    pub first_blocknum: u32,
+
+    /// number of blocks
+    pub number_of_blocks: u32,
+
+    /// full information
+    pub full_information: u8,
+
+    /// encoded short ids
+    pub encoded_short_ids: Vec<ShortChannelId>,
+
+    /// reply channel range tlvs
+    #[tlv(type = 1)]
+    pub reply_channel_range_tlvs: BTreeMap<u8, Vec<u8>>,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct GossipTimestampFilter {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// first timestamp
+    pub first_timestamp: u32,
```

Comment:
Also considering introducing type `LightningTimestamp` as a wrapper for `u32` to be used everywhere in the messages here - with custom Display & Debug implementations, conversion to normal timestamps etc. But this can be an issue for the future work, like with bitflags and other custom types.

Reply:
Creating an issue for future work.